packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -109,7 +116,105 @@ export default createRule<
     const seenLogicals = new Set<TSESTree.LogicalExpression>();
 
     return {
-      // specific handling for `(foo ?? {}).bar` / `(foo || {}).bar`
+      // specific handling for `if (foo) { foo.bar(); }` / `if (foo) foo.bar();`
+      'IfStatement[consequent.type=BlockStatement][consequent.body.length=1], IfStatement[consequent.type=ExpressionStatement]':
+        (
+          node: {
+            consequent: {
+              type:
+                | AST_NODE_TYPES.BlockStatement
+                | AST_NODE_TYPES.ExpressionStatement;
+            };
+          } & TSESTree.IfStatement,
+        ): void => {
+          if (!options.allowSuggestingOnIfStatements) {
+            return;
+          }
+          if (node.alternate) {
```

Comment:
[Style] Nit: we can save a bit of space...

```suggestion
          if (!options.allowSuggestingOnIfStatements || node.alternate) {
```

Reply:
I usually nitpick about this as well in PRs, but here I wanted the option toggling to be separate from the rule logic, but I don't mind that much.